### PR TITLE
Modification in Collect() method of GenericCollector

### PR DESCRIPTION
**Pull request by anushree-n**
Calculation of minimum polling frequency and check for validity of regular expressions of all metrics of a collector has been done during initialization of the collector.

### CODE REVIEW

File: collector/generic_collector.go [hunk 1]
```diff
@@ -95,14 +95,8 @@ func (collector *GenericCollector) Name() string {
...
 func (collector *GenericCollector) Collect() (time.Time, []v1.Metric, error) {
-	minNextColTime := collector.configFile.MetricsConfig[0].PollingFrequency
-	for _, metricConfig := range collector.configFile.MetricsConfig {
-		if metricConfig.PollingFrequency < minNextColTime {
-			minNextColTime = metricConfig.PollingFrequency
-		}
-	}
 	currentTime := time.Now()
-	nextCollectionTime := currentTime.Add(time.Duration(minNextColTime * time.Second))
+	nextCollectionTime := currentTime.Add(time.Duration(collector.info.minPollingFrequency * time.Second))
```
Comment (rjnagal) [inline collector/generic_collector.go]: In line 68: It looks like we don't check if the user provided a polling frequency for any metric config. We need to set in a default if json returns zero and ignore any metricConfig.PollingFrequency which is zero.

Comment (anushree-n) [inline collector/generic_collector.go]: Added the change. :)

File: collector/generic_collector.go [hunk 2]
```diff
@@ -78,6 +78,10 @@ func NewCollector(collectorName string, configfile string) (*GenericCollector, e
 			return nil, fmt.Errorf("Invalid regexp %v for metric %v", metricConfig.Regex, metricConfig.Name)
 		}
 	}
+	//set minPollFrequency to maxHousekeepingInterval if config returns minpollFrequency=0
+	if minPollFrequency == 0 {
+		minPollFrequency = 60 * time.Second
```
Comment (rjnagal) [inline collector/generic_collector.go]: Do we want the default to be one minute? The default housekeeping is 1s.

File: collector/generic_collector.go [hunk 3]
```diff
@@ -78,6 +78,10 @@ func NewCollector(collectorName string, configfile string) (*GenericCollector, e
 			return nil, fmt.Errorf("Invalid regexp %v for metric %v", metricConfig.Regex, metricConfig.Name)
 		}
 	}
+	//set minPollFrequency to maxHousekeepingInterval if config returns minpollFrequency=0
+	if minPollFrequency == 0 {
```
Comment (rjnagal) [inline collector/generic_collector.go]: I think we should move this logic up to replace line 68. Start with the default and then compare in the loop (#72) and lower it if we find a user-defined value.

**Comment by rjnagal** [comment]
LGTM

**Pull request merged**